addons/addon-base-raas/packages/base-raas-services/lib/user/__tests__/user-authz-service.test.js

```diff
@@ -186,40 +186,84 @@ describe('UserAuthzService', () => {
       projectId: 'Rokovoko',
-      randomAttrib: 'Kujira',
+      lastName: 'Kujira',
       userType: 'tree',
     };
     const existingRootUser = {
-      userRole: 'Suifu',
+      userRole: 'admin',
       authenticationProviderId: 'Fedallah',
       projectId: 'Tashtego',
-      randomAttrib: 'Daggoo',
+      lastName: 'Daggoo',
       userType: 'root',
     };
+    const existingAdminUser = {
+      userRole: 'admin',
+      isAdmin: true,
+      authenticationProviderId: 'Fedallah',
+      projectId: ['Tashtego'],
+      lastName: 'Daggoo',
+    };
 
-    it('should only allow users to change unprotected attributes of other users', async () => {
+    it('should only allow users to change unprotected attributes of themselves', async () => {
       // BUILD
+      const isExternalUser = {
+        isExternalUser: true,
+      };
+      const isAdmin = {
+        isAdmin: true,
+      };
+      const projectId = {
+        projectId: ['testProjectId1', 'testProjectId2'],
+      };
+      const identityProviderName = {
+        identityProviderName: 'some-identity-provider-name',
+      };
+      const authenticationProviderId = {
+        authenticationProviderId: '123456',
+      };
+      const userType = {
+        userType: 'root',
+      };
 
       // OPERATE
       const userUserUnprot = await service.authorizeUpdateAttributes({}, {}, unprotAttr, existingUser);
-      const userUserProt = await service.authorizeUpdateAttributes({}, {}, protAttr, existingUser);
       const userRootUnprot = await service.authorizeUpdateAttributes({}, {}, unprotAttr, existingRootUser);
+      const userUpdateRole = await service.authorizeUpdateAttributes({}, {}, userRole, existingUser);
+      const userUpdateIsExternalUser = await service.authorizeUpdateAttributes({}, {}, isExternalUser, existingUser);
+      const userUpdateIsAdmin = await service.authorizeUpdateAttributes({}, {}, isAdmin, existingUser);
+      const userUpdateProjectId = await service.authorizeUpdateAttributes({}, {}, projectId, existingUser);
+      const userUpdateIdpName = await service.authorizeUpdateAttributes({}, {}, identityProviderName, existingUser);
+      const userUpdateIdpId = await service.authorizeUpdateAttributes({}, {}, authenticationProviderId, existingUser);
+      const userUpdateType = await service.authorizeUpdateAttributes({}, {}, userType, existingUser);
+      const adminUpdateType = await service.authorizeUpdateAttributes(
+        { principal: { isAdmin: true } },
+        {},
+        userType,
+        existingAdminUser,
+      );
 
       // CHECK
       expect(userUserUnprot).toMatchObject({ effect: 'allow' });
-      expect(userUserProt).toMatchObject({ effect: 'deny' });
       expect(userRootUnprot).toMatchObject({ effect: 'deny' });
+      expect(userUpdateRole).toMatchObject({ effect: 'deny' });
+      expect(userUpdateIsExternalUser).toMatchObject({ effect: 'deny' });
+      expect(userUpdateIsAdmin).toMatchObject({ effect: 'deny' });
```

Comment:
Can we add a negative test here which tests that admin can update the isAdmin flag